[atem] Add dtype check to bmm_out_or_baddbmm_

hlu1 opened the pull request:
Summary: Fix bug raised in https://github.com/pytorch/pytorch/issues/50980 by adding a dtype check in bmm_out_or_baddbmm_.

Test Plan:
```
buck test //caffe2/test:linalg
buck test //caffe2/aten:math_kernel_test
```

Differential Revision: D26113575

ngimel commented on aten/src/ATen/native/LinearAlgebra.cpp:
```diff
@@ -731,6 +731,13 @@ static inline Tensor& bmm_out_or_baddbmm_(Tensor& self_or_result, const Tensor&
   checkOnCPU(batch1, c);
   checkOnCPU(batch2, c);
 
+  TORCH_CHECK(
```
can you instead change accesses to .data_ptr<scalar_t>() in mkl/LinearAlgebra.cpp? Otherwise, baddbmm_mkl_template function is needlessly unsafe. You also need to either change `res.data_ptr<scalar_t>()` there (preferably), or check self_or_result dtype here.

hlu1 replied:
Fixed

Commit:
[atem] Add dtype check to bmm_out_or_baddbmm_ (#51248)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/51248

Fix bug raised in https://github.com/pytorch/pytorch/issues/50980 by adding a dtype check in bmm_out_or_baddbmm_.

Test Plan:
```
buck test //caffe2/test:linalg
buck test //caffe2/aten:math_kernel_test
```

Differential Revision: D26113575

fbshipit-source-id: 03c5ad579c812d2745f060b32e85543793e2ba29

ngimel reviewed (approved):
Thanks for the fix!
